tests/nvmeof/test_ceph_nvmeof_qos_tests.py

```diff
@@ -0,0 +1,344 @@
+        config: test config
+    """
+    # Delete the multiple Initiators across multiple gateways
+    if "initiators" in config["cleanup"]:
+        for initiator_cfg in config["initiators"]:
+            disconnect_initiator(
+                ceph_cluster, initiator_cfg["node"], initiator_cfg["nqn"]
+            )
+
+    if "disconnect_all" in config["cleanup"]:
+        nodes = config.get("disconnect_all")
+        if not nodes:
+            nodes = [i["node"] for i in config["initiators"]]
+
+        disconnect_all_initiator(ceph_cluster, nodes)
+
+    # Delete the multiple subsystems across multiple gateways
+    if "subsystems" in config["cleanup"]:
+        config_sub_node = config["subsystems"]
+        if not isinstance(config_sub_node, list):
+            config_sub_node = [config_sub_node]
+        for sub_cfg in config_sub_node:
+            node = config["gw_node"] if "node" not in sub_cfg else sub_cfg["node"]
+            LOG.info(f"Deleting subsystem {sub_cfg['nqn']} on gateway {node}")
+            nvmegwcli.subsystem.delete(
+                **{"args": {"subsystem": sub_cfg["nqn"], "force": True}}
+            )
+
+    # Delete the gateway
+    if "gateway" in config["cleanup"]:
+        delete_nvme_service(ceph_cluster, config)
+
+    # Delete the pool
+    if "pool" in config["cleanup"]:
+        rbd_obj.clean_up(pools=[config["rbd_pool"]])
+
+
+def trigger_fio(ceph_cluster, config, io_mode, key):
+    results = []
+    client = get_node_by_id(ceph_cluster, config["initiators"][0]["node"])
+    initiator = Initiator(client)
+
+    try:
+        # List NVMe targets
+        targets = initiator.list_spdk_drives()
+        if not targets:
+            raise Exception(f"NVMe Targets not found on {client.hostname}")
+        LOG.debug(targets)
+
+        with parallel() as p:
+            for target in targets:
+                try:
+                    _io_args = {
+                        "device_name": target["DevicePath"],
+                        "client_node": client,
+                        "long_running": True,
+                        "io_type": io_mode,
+                        "run_time": "50",
+                        "size": "100%",
+                        "iodepth": 4,
+                        "time_based": True,
+                    }
+                    if key == "rw-ios-per-second":
+                        _io_args.update({"rwmixread": "50"})
+                    p.spawn(run_fio, **_io_args)
+                except Exception as fio_err:
+                    LOG.error(
+                        f"Error running FIO on target {target['DevicePath']}: {fio_err}"
+                    )
+                    results.append(f"Failed: {fio_err}")
+
+            for op in p:
+                results.append(op)
+
+    except Exception as err:
+        LOG.error(f"Error in trigger_fio: {err}")
+        raise
+
+    return results
+
+
+def run_io_and_validate_qos(ceph_cluster, config, key, qos_args, nvmegwcli):
```

Comment:
I would suggest to move QOS definitions to new module called qos under workflows, Since we have lot of things to validate and keeping test module to call different scenarios around QOS.